cmd/viam-agent/main.go

```diff
@@ -47,7 +46,6 @@ type agentOpts struct {
 //nolint:gocognit
 func commonMain() {
 	ctx, cancel := setupExitSignalHandling()
-	globalCancel = cancel
```

Comment:
This was the only place this was ever set. Without it, the _windows file will always get a nil. I made a few suggestions above/below here which fixes the linter's problem (where it deletes the nolint, then complains about it an unused variable.

Reply:
tbh i was little confused on how this was working but i get it now